[export] Allow constant outputs + None input/outputs

angelayi opened the pull request:
Added support for constant outputs. We will just embed the constant directly into the output, like `return (x, 1)`.
Also adds support for None input/outputs. For None inputs we address it the same way we do to constants, which is that a placeholder with no users will be inserted into the graph, and the None will be embedded into whatever operator is using the None. For None outputs, we will also address the same way we do constants, which is that we embed it into the output, like `return (x, None)`.

Differential Revision: D52881070



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

avikchaudhuri reviewed (commented):
Cool! Wonder if it's worth writing a test showing the graph for constant outputs.

What's up with the failing tests, though? Look related...

zhxchen17 commented:
@angelayi looks fine to me! All the failing tests should be fixed because they are all assertRaisesRegex.

angelayi committed:
[export] Allow constant outputs + None input/outputs (#117894)

Summary:

Added support for constant outputs, which is that we just embed the constant directly into the output, like `return (x, 1)`.
Also adds support for None input/outputs. For None inputs we address it the same way we do to constants, which is that a placeholder with no users will be inserted into the graph, and the None will be embedded into whatever operator is using the None.

Test Plan: CI

Reviewed By: zhxchen17

Differential Revision: D52881070
